rfc/rfc007-api-stability.md

```diff
@@ -0,0 +1,93 @@
+---
+RFC: 007
+Author: Andrew Gerrand <adg@golang.org>
+Status: Draft
+---
+
+# API Stability Policy
+
+## Motivation
+
+The gokit project depends on code maintained by others.
+This includes the Go standard library and sub-repositories
+and other external libraries.
+
+The Go project provides stability guarantees, but the other external libraries
+typically do not. This RFC proposes a standard policy for package authors to
+advertise API stability.
+
+The intention is that the gokit project will require that its dependencies
+adhere to the policy, with the greater goal of improving the Go ecosystem
+as a whole.
+
+## Scope
+
+This policy is for package authors to provide their users with a promise of API
+stability.
+This document is similar to and inspired by the [Go 1 compatibility
+promise](https://golang.org/doc/go1compat).
+
+An author declaring their package "API Stable" makes the following promise:
+
+> We will not change the package's exported API in backward incompatible ways.
+> Future changes to this package will not break dependent code.
+
+### Coverage
+
+The promise of stability includes:
+
+* The package name,
+* Exported type declarations and struct fields (names and types),
+* Exported function and method names, parameters, and return values,
```

Comment:
Well, this bullet point talks about functions and methods, which is why I was only focused on that. But yes, exported vars and typed consts should probably have to have an exported or builtin type too.

Just ruling out mixing exported and unexported names in function/method/type/var/const declarations might be sufficient; it can be weakened in the future.